### PR TITLE
chore: update test environment

### PR DESCRIPTION
**Pull request by bjornstar**
- start testing on node 20
- drop testing on node 16
- update `ava` from `^5.1.0` to `6.1.3`
- update `xo` from `^v0.53.1` to `v0.58.0`
- add todo for new lint rule (followup PR coming 😉)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ https.get(url, response => {
 });
 ```
 */
+// eslint-disable-next-line @typescript-eslint/ban-types -- TODO: Switch to Uint8Array only
```
Comment (sindresorhus) [inline index.d.ts:71]: Removing Buffer here is not a breaking change and should just be done.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -10,8 +10,8 @@ jobs:
       fail-fast: false
       matrix:
         node-version:
+          - 20
           - 18
-          - 16
```
Comment (sindresorhus) [inline .github/workflows/main.yml:14]: 16 should be preserved

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (bjornstar) [reply .github/workflows/main.yml:14]: I have a question, [you made node 18 the minimum required version](https://github.com/sindresorhus/file-type/commit/7f4b30b2ef89f73279fb4ff95e5c38e9a0687181) for `file-type` which this library is a thin wrapper for. Would a followup be welcome to update to the latest major version of `file-type` and drop node 16 support?

Comment (sindresorhus) [reply .github/workflows/main.yml:14]: Yes

**Pull request merged**